### ISSUE TITLE
Issue with installing extensions

### ISSUE THREAD

**Issue by roshanjonah** [sherifabdlnaby/kubephp#12]
I got the following error
```
 > [app 6/6] RUN post-build-base && post-build-prod:
#30 0.177 ► Running Composer Install...
#30 0.284 Installing dependencies from lock file
#30 0.286 Verifying lock file contents can be installed on current platform.
#30 0.306 Your lock file does not contain a compatible set of packages. Please run composer update.
#30 0.306
#30 0.306   Problem 1
#30 0.306     - Root composer.json requires PHP extension ext-zip * but it is missing from your system. Install or enable PHP's zip extension.
#30 0.306   Problem 2
#30 0.306     - Root composer.json requires PHP extension ext-pcntl * but it is missing from your system. Install or enable PHP's pcntl extension.
#30 0.306   Problem 3
#30 0.306     - Root composer.json requires PHP extension ext-imap * but it is missing from your system. Install or enable PHP's imap extension.
#30 0.306   Problem 4
#30 0.306     - Root composer.json requires PHP extension ext-intl * but it is missing from your system. Install or enable PHP's intl extension.
#30 0.306   Problem 5
#30 0.306     - Root composer.json requires PHP extension ext-tidy * but it is missing from your system. Install or enable PHP's tidy extension.
#30 0.306   Problem 6
#30 0.306     - 2amigos/qrcode-library is locked to version 1.1.2 and an update of this package was not requested.
#30 0.306     - 2amigos/qrcode-library 1.1.2 requires ext-gd * -> it is missing from your system. Install or enable PHP's gd extension.
#30 0.306   Problem 7
#30 0.306     - php-amqplib/php-amqplib is locked to version v2.12.3 and an update of this package was not requested.
#30 0.306     - php-amqplib/php-amqplib v2.12.3 requires ext-sockets * -> it is missing from your system. Install or enable PHP's sockets extension.
#30 0.306
#30 0.306 To enable extensions, verify that they are enabled in your .ini files:
#30 0.306     - /usr/local/etc/php/php.ini
#30 0.306     - /usr/local/etc/php/conf.d/base-php.ini
#30 0.306     - /usr/local/etc/php/conf.d/docker-php-ext-sodium.ini
#30 0.306     - /usr/local/etc/php/conf.d/prod-php.ini
#30 0.306 You can also run `php --ini` inside terminal to see which files are used by PHP in CLI mode.
```
My Dockerfile contains this section and everything else is default config

```
...
 # PHP Extensions --------------------------------- \
 && docker-php-ext-install -j$(nproc) \
      intl        \
      opcache     \
      pdo_mysql   \
      zip         \
      mysql       \
      bcmath      \
      bz2         \
      gd          \
      gmp         \
      imap        \
      ldap        \
      tidy        \
      xsl         \
      curl        \
      json        \
      dev         \
      mbstring    \
      xml         \
      pcntl       \
...
```

When I run `make build` it throws out that error. What could be the issue? Any help would be appreciated. It's a Yii framework PHP app.

**Comment by sherifabdlnaby** [comment]
Judging by the logs here
```To enable extensions, verify that they are enabled in your .ini files:
#30 0.306     - /usr/local/etc/php/php.ini
#30 0.306     - /usr/local/etc/php/conf.d/base-php.ini
#30 0.306     - /usr/local/etc/php/conf.d/docker-php-ext-sodium.ini
#30 0.306     - /usr/local/etc/php/conf.d/prod-php.ini
```
It seems like the Install Command failed to install the extensions (yet didn't fail because it passed the stage), because otherwise, you should have a `docker-php-ext-$extensionName.ini` files listed.

Can you run the make build with `DOCKER_BUILDKIT=1 COMPOSE_DOCKER_CLI_BUILD=1 BUILDKIT_PROGRESS=plain docker-compose build --no-cache` and post the build logs especially at the part when the extensions are installed?

**Comment by roshanjonah** [comment]
Thank you @sherifabdlnaby for looking into this. I am a newbie so chances are maybe I am doing something wrong here. 

```
#9 [base  3/12] RUN apk add --no-cache --virtual .build-deps       autoconf 		dpkg-dev dpkg 		file 		g++ 		gcc 		libc-dev 		make 		pkgconf 	re2c        libzip-dev          icu-dev        && docker-php-ext-install -j$(nproc)       intl              opcache           pdo_mysql         zip               mysql             bcmath            bz2               gd                gmp               imap              ldap              tidy              xsl               curl              json              dev               mbstring          xml               pcntl        && pecl install apcu-5.1.20 && docker-php-ext-enable apcu zip  && pecl install xdebug-${XDEBUG_VERSION}  && runDeps="$( 		scanelf --needed --nobanner --format '%n#p' --recursive /usr/local/lib/php/extensions 			| tr ',' '\n' 			| sort -u 			| awk 'system("[ -e /usr/local/lib/" $1 " ]") == 0 { next } { print "so:" $1 }' 	)"; 	apk add --no-network --virtual .php-extensions-rundeps $runDeps; 	apk del --no-network .build-deps
#9 sha256:3e9e30d85b5f4683bcd27906a3d7e069d5f67740aaf160753ea016718fcdc9b2
#9 0.342 fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/main/x86_64/APKINDEX.tar.gz
#9 0.447 fetch https://dl-cdn.alpinelinux.org/alpine/v3.14/community/x86_64/APKINDEX.tar.gz
#9 0.646 (1/32) Installing m4 (1.4.18-r2)
#9 0.667 (2/32) Installing libbz2 (1.0.8-r1)
#9 0.678 (3/32) Installing perl (5.32.1-r0)
#9 1.402 (4/32) Installing autoconf (2.71-r0)
#9 1.456 (5/32) Installing pkgconf (1.7.4-r0)
#9 1.468 (6/32) Installing dpkg-dev (1.20.9-r0)
#9 1.503 (7/32) Installing dpkg (1.20.9-r0)
#9 1.545 (8/32) Installing libmagic (5.40-r1)
#9 1.588 (9/32) Installing file (5.40-r1)
#9 1.597 (10/32) Installing libgcc (10.3.1_git20210424-r2)
#9 1.609 (11/32) Installing libstdc++ (10.3.1_git20210424-r2)
#9 1.660 (12/32) Installing binutils (2.35.2-r2)
#9 1.884 (13/32) Installing libatomic (10.3.1_git20210424-r2)
#9 1.894 (14/32) Installing libgphobos (10.3.1_git20210424-r2)
#9 2.102 (15/32) Installing gmp (6.2.1-r0)
#9 2.135 (16/32) Installing isl22 (0.22-r0)
#9 2.258 (17/32) Installing mpfr4 (4.1.0-r0)
#9 2.444 (18/32) Installing mpc1 (1.2.1-r0)
#9 2.461 (19/32) Installing gcc (10.3.1_git20210424-r2)
#9 5.851 (20/32) Installing musl-dev (1.2.2-r3)
#9 6.220 (21/32) Installing libc-dev (0.7.2-r3)
#9 6.237 (22/32) Installing g++ (10.3.1_git20210424-r2)
#9 8.360 (23/32) Installing make (4.3-r0)
#9 8.381 (24/32) Installing re2c (2.1.1-r0)
#9 8.428 (25/32) Installing zlib-dev (1.2.11-r3)
#9 8.449 (26/32) Installing xz-dev (5.2.5-r0)
#9 8.471 (27/32) Installing libzip (1.7.3-r2)
#9 8.490 (28/32) Installing libzip-dev (1.7.3-r2)
#9 8.499 (29/32) Installing icu-libs (67.1-r2)
#9 9.838 (30/32) Installing icu (67.1-r2)
#9 9.864 (31/32) Installing icu-dev (67.1-r2)
#9 9.962 (32/32) Installing .build-deps (20211122.045720)
#9 9.964 Executing busybox-1.33.1-r6.trigger
#9 9.971 OK: 288 MiB in 70 packages
#9 11.58 error: /usr/src/php/ext/mysql does not exist
#9 11.58
#9 11.58 usage: /usr/local/bin/docker-php-ext-install [-jN] [--ini-name file.ini] ext-name [ext-name ...]
#9 11.58    ie: /usr/local/bin/docker-php-ext-install gd mysqli
#9 11.58        /usr/local/bin/docker-php-ext-install pdo pdo_mysql
#9 11.58        /usr/local/bin/docker-php-ext-install -j5 gd mbstring mysqli pdo pdo_mysql shmop
#9 11.58
#9 11.58 if custom ./configure arguments are necessary, see docker-php-ext-configure
#9 11.58
#9 11.58 Possible values for ext-name:
#9 11.61 bcmath bz2 calendar ctype curl dba dom enchant exif fileinfo filter ftp gd gettext gmp hash iconv imap interbase intl json ldap mbstring mysqli oci8 odbc opcache pcntl pdo pdo_dblib pdo_firebird pdo_mysql pdo_oci pdo_odbc pdo_pgsql pdo_sqlite pgsql phar posix pspell readline recode reflection session shmop simplexml snmp soap sockets sodium spl standard sysvmsg sysvsem sysvshm tidy tokenizer wddx xml xmlreader xmlrpc xmlwriter xsl zend_test zip
#9 11.61
#9 11.61 Some of the above modules are already compiled into PHP; please check
#9 11.61 the output of "php -i" to see which modules are already loaded.
#9 11.62 WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.14/main: No such file or directory
#9 11.62 WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.14/community: No such file or directory
#9 11.62 (1/1) Installing .php-extensions-rundeps (20211122.045731)
#9 11.62 OK: 288 MiB in 71 packages
#9 11.63 WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.14/main: No such file or directory
#9 11.63 WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.14/community: No such file or directory
#9 11.63 (1/32) Purging .build-deps (20211122.045720)
#9 11.63 (2/32) Purging autoconf (2.71-r0)
#9 11.63 (3/32) Purging m4 (1.4.18-r2)
#9 11.63 (4/32) Purging dpkg-dev (1.20.9-r0)
#9 11.64 (5/32) Purging perl (5.32.1-r0)
#9 11.66 (6/32) Purging dpkg (1.20.9-r0)
#9 11.66 (7/32) Purging file (5.40-r1)
#9 11.66 (8/32) Purging g++ (10.3.1_git20210424-r2)
#9 11.69 (9/32) Purging gcc (10.3.1_git20210424-r2)
#9 11.71 (10/32) Purging binutils (2.35.2-r2)
#9 11.71 (11/32) Purging libatomic (10.3.1_git20210424-r2)
#9 11.71 (12/32) Purging libgphobos (10.3.1_git20210424-r2)
#9 11.71 (13/32) Purging libc-dev (0.7.2-r3)
#9 11.71 (14/32) Purging musl-dev (1.2.2-r3)
#9 11.72 (15/32) Purging make (4.3-r0)
#9 11.72 (16/32) Purging re2c (2.1.1-r0)
#9 11.72 (17/32) Purging libzip-dev (1.7.3-r2)
#9 11.72 (18/32) Purging zlib-dev (1.2.11-r3)
#9 11.72 (19/32) Purging xz-dev (5.2.5-r0)
#9 11.72 (20/32) Purging libzip (1.7.3-r2)
#9 11.72 (21/32) Purging icu-dev (67.1-r2)
#9 11.73 (22/32) Purging icu (67.1-r2)
#9 11.73 (23/32) Purging icu-libs (67.1-r2)
#9 11.73 (24/32) Purging libbz2 (1.0.8-r1)
#9 11.73 (25/32) Purging pkgconf (1.7.4-r0)
#9 11.73 (26/32) Purging libmagic (5.40-r1)
#9 11.73 (27/32) Purging libstdc++ (10.3.1_git20210424-r2)
#9 11.73 (28/32) Purging libgcc (10.3.1_git20210424-r2)
#9 11.73 (29/32) Purging mpc1 (1.2.1-r0)
#9 11.73 (30/32) Purging mpfr4 (4.1.0-r0)
#9 11.73 (31/32) Purging isl22 (0.22-r0)
#9 11.73 (32/32) Purging gmp (6.2.1-r0)
#9 11.74 Executing busybox-1.33.1-r6.trigger
#9 11.74 OK: 17 MiB in 39 packages
#9 DONE 12.3s
```

**Comment by sherifabdlnaby** [comment]
Use correct extension name
```Possible values for ext-name:
#9 11.61 bcmath bz2 calendar ctype curl dba dom enchant exif fileinfo filter ftp gd gettext gmp hash iconv imap interbase intl json ldap mbstring mysqli oci8 odbc opcache pcntl pdo pdo_dblib pdo_firebird pdo_mysql pdo_oci pdo_odbc pdo_pgsql pdo_sqlite pgsql phar posix pspell readline recode reflection session shmop simplexml snmp soap sockets sodium spl standard sysvmsg sysvsem sysvshm tidy tokenizer wddx xml xmlreader xmlrpc xmlwriter xsl zend_test zip
```

**Comment by roshanjonah** [comment]
My extension list looks like this now
```
...
 # PHP Extensions --------------------------------- \
 && docker-php-ext-install -j$(nproc) \
      intl        \
      opcache     \
      pdo_mysql   \
      zip         \
      mysqli      \
      bcmath      \
      bz2         \
      gd          \
      gmp         \
      imap        \
      ldap        \
      tidy        \
      xsl         \
      curl        \
      json        \
      mbstring    \
      xml         \
...
```

But I am still experiencing errors when doing "make build" or "make up"

```
...
 => CACHED [app-dev 1/6] RUN apk add git openssh                                                                                                                                                             0.0s
 => ERROR [app-dev 2/6] RUN docker-php-ext-enable xdebug                                                                                                                                                     0.2s
------
 > [app-dev 2/6] RUN docker-php-ext-enable xdebug:
#20 0.185 PHP Warning:  PHP Startup: Unable to load dynamic library 'intl.so' (tried: /usr/local/lib/php/extensions/no-debug-non-zts-20180731/intl.so (Error loading shared library libicuio.so.67: No such file or directory (needed by /usr/local/lib/php/extensions/no-debug-non-zts-20180731/intl.so)), /usr/local/lib/php/extensions/no-debug-non-zts-20180731/intl.so.so (Error loading shared library /usr/local/lib/php/extensions/no-debug-non-zts-20180731/intl.so.so: No such file or directory)) in Unknown on line 0
#20 0.185 PHP Warning:  PHP Startup: Unable to load dynamic library 'zip.so' (tried: /usr/local/lib/php/extensions/no-debug-non-zts-20180731/zip.so (Error loading shared library libzip.so.5: No such file or directory (needed by /usr/local/lib/php/extensions/no-debug-non-zts-20180731/zip.so)), /usr/local/lib/php/extensions/no-debug-non-zts-20180731/zip.so.so (Error loading shared library /usr/local/lib/php/extensions/no-debug-non-zts-20180731/zip.so.so: No such file or directory)) in Unknown on line 0
#20 0.190 error: 'xdebug' does not exist
#20 0.190
#20 0.190 usage: /usr/local/bin/docker-php-ext-enable [options] module-name [module-name ...]
#20 0.190    ie: /usr/local/bin/docker-php-ext-enable gd mysqli
#20 0.190        /usr/local/bin/docker-php-ext-enable pdo pdo_mysql
#20 0.190        /usr/local/bin/docker-php-ext-enable --ini-name 0-apc.ini apcu apc
#20 0.190
#20 0.190 Possible values for module-name:
#20 0.194 bcmath.so intl.so mysqli.so opcache.so pdo_mysql.so sodium.so zip.so
#20 0.194
#20 0.194 Some of the above modules are already compiled into PHP; please check
#20 0.194 the output of "php -i" to see which modules are already loaded.
------
executor failed running [/bin/sh -c docker-php-ext-enable xdebug]: exit code: 1
ERROR: Service 'app' failed to build : Build failed
make: *** [up] Error 1
```

What do you think @sherifabdlnaby

**Comment by roshanjonah** [comment]
Now I can't see any error in extension installation but it says the following:

```
#28 [app 6/6] RUN post-build-base && post-build-prod
#28 sha256:8c029f7f720b07dd485668efea70a2a583e523d3f63dd02b97748b2d1ac082a4
#28 0.272 ► Running Composer Install...
#28 0.278 PHP Warning:  PHP Startup: Unable to load dynamic library 'zip.so' (tried: /usr/local/lib/php/extensions/no-debug-non-zts-20190902/zip.so (Error loading shared library libzip.so.5: No such file or directory (needed by /usr/local/lib/php/extensions/no-debug-non-zts-20190902/zip.so)), /usr/local/lib/php/extensions/no-debug-non-zts-20190902/zip.so.so (Error loading shared library /usr/local/lib/php/extensions/no-debug-non-zts-20190902/zip.so.so: No such file or directory)) in Unknown on line 0
#28 0.375 Installing dependencies from lock file
#28 0.377 Verifying lock file contents can be installed on current platform.
#28 0.400 Your lock file does not contain a compatible set of packages. Please run composer update.
#28 0.400
#28 0.400   Problem 1
#28 0.400     - Root composer.json requires PHP extension ext-zip * but it is missing from your system. Install or enable PHP's zip extension.
#28 0.400   Problem 2
#28 0.400     - Root composer.json requires PHP extension ext-pcntl * but it is missing from your system. Install or enable PHP's pcntl extension.
#28 0.400   Problem 3
#28 0.400     - Root composer.json requires PHP extension ext-imap * but it is missing from your system. Install or enable PHP's imap extension.
#28 0.400   Problem 4
#28 0.400     - Root composer.json requires PHP extension ext-intl * but it is missing from your system. Install or enable PHP's intl extension.
#28 0.400   Problem 5
#28 0.400     - Root composer.json requires PHP extension ext-tidy * but it is missing from your system. Install or enable PHP's tidy extension.
#28 0.400   Problem 6
#28 0.400     - 2amigos/qrcode-library is locked to version 1.1.2 and an update of this package was not requested.
#28 0.400     - 2amigos/qrcode-library 1.1.2 requires ext-gd * -> it is missing from your system. Install or enable PHP's gd extension.
#28 0.400   Problem 7
#28 0.400     - php-amqplib/php-amqplib is locked to version v2.12.3 and an update of this package was not requested.
#28 0.400     - php-amqplib/php-amqplib v2.12.3 requires ext-sockets * -> it is missing from your system. Install or enable PHP's sockets extension.
#28 0.400
#28 0.400 To enable extensions, verify that they are enabled in your .ini files:
#28 0.400     - /usr/local/etc/php/php.ini
#28 0.400     - /usr/local/etc/php/conf.d/base-php.ini
#28 0.400     - /usr/local/etc/php/conf.d/docker-php-ext-bcmath.ini
#28 0.400     - /usr/local/etc/php/conf.d/docker-php-ext-mysqli.ini
#28 0.400     - /usr/local/etc/php/conf.d/docker-php-ext-opcache.ini
#28 0.400     - /usr/local/etc/php/conf.d/docker-php-ext-pdo_mysql.ini
#28 0.400     - /usr/local/etc/php/conf.d/docker-php-ext-sodium.ini
#28 0.400     - /usr/local/etc/php/conf.d/docker-php-ext-zip.ini
#28 0.400     - /usr/local/etc/php/conf.d/prod-php.ini
#28 0.400 You can also run `php --ini` inside terminal to see which files are used by PHP in CLI mode.
#28 ERROR: executor failed running [/bin/sh -c post-build-base && post-build-prod]: exit code: 2
------
 > [app 6/6] RUN post-build-base && post-build-prod:
------
executor failed running [/bin/sh -c post-build-base && post-build-prod]: exit code: 2
ERROR: Service 'app' failed to build : Build failed
```

But I have put it down to be installed. Do you have any other way to troubleshoot this? @sherifabdlnaby

**Comment by sherifabdlnaby** [comment]
@roshanjonah I pushed a fix at `v4.1.1` please try sync'ing your changes and build again.
